plutus-tx/src/PlutusTx/Builtins.hs

```diff
@@ -202,6 +215,276 @@ verifySchnorrSecp256k1Signature
 verifySchnorrSecp256k1Signature vk msg sig =
   fromBuiltin (BI.verifySchnorrSecp256k1Signature vk msg sig)
 
+-- | Converts an 'Integer' into its 'BuiltinByteString' representation.
+--
+-- = Notes
+--
+-- Throughout, let @maxInteger@ be the maximum function for 'Integer's,
+-- and @absInteger@ be the absolute value function for 'Integer's.
+-- We define @zeroes :: 'Integer' -> 'BuiltinByteString'@ be the
+-- function which, given input @i@, produces a 'BuiltinByteString' @bs@ such
+-- that:
+--
+-- * @'lengthByteString' bs@ @=@ @'maxInteger' 0 i@; and
+-- * For all @j :: 'Integer'@ such that @'greaterThanEqualsInteger' j 0@,
+-- @'indexByteString' bs j = 0@;
+--
+-- We define @ones :: 'Integer' -> 'BuiltinByteString'@ identically to @zeroes@,
+-- except that @'indexByteString' bs j = 255@ instead.
+--
+-- == Laws
+--
+-- 'integerToByteString' must roundtrip via 'byteStringToInteger'. Specifically,
```

Comment:
Do you want to observe that the opposite direction holds up to the representation of 0?

Reply:
Ah, you mention it below.